### PR TITLE
feat: update to tree-sitter 0.23

### PR DESCRIPTION
**Pull request by hendrikvanantwerpen**
Tree-sitter released version 0.23 recently. This contains a breaking, but ultimately very good change to how parser bindings work (https://github.com/tree-sitter/tree-sitter/pull/3069). The TLDR of that change is that parsers do not depend on tree-sitter anymore, but instead on a shard (and supposedly very stable) tree-sitter-language crate. As a result, clients are free to chose their tree-sitter version as the parser ABI is supported. Library and parser versions are less tighly coupled, and it should no longer be necessary to move all the parsers to the next tree-sitter version in lock-step to be able to upgrade the library.

Most of the changes in this PR are from running tree-sitter generate. I'll add comments on other changes I did to explain why I thought they are necessary, or where I'm not sure I did it correctly.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,8 +38,8 @@
     }
   },
   "scripts": {
-    "build": "tree-sitter generate --no-bindings",
-    "test": "tree-sitter test",
+    "build": "npx tree-sitter-cli generate --no-bindings",
+    "test": "npx tree-sitter-cli test",
```
Comment (hendrikvanantwerpen) [inline package.json:42]: The tree-sitter invocations were changed to make the scripts a bit more robust. Calling tree-sitter through npx like this ensures the correct version from the dev dependencies is used, and makes the scripts independent from the local system state.

**Comment by hendrikvanantwerpen** [comment]
@amaanq I saw you did similar changes to many of the grammars in the tree-sitter org, so perhaps it is possible that you review this?

**Comment by amaanq** [comment]
Yeah I can take a look & tweak this later today.

File: package.json (hunk 1 above)
Comment (ObserverOfTime) [reply package.json:42]: npm prefers binaries in node_modules anyway.

Comment (hendrikvanantwerpen) [reply package.json]: > npm prefers binaries in node_modules anyway.

Ah, that's good to know. I've reverted this change then.

**Review by ObserverOfTime — APPROVED** [review]

**Pull request merged**